node/actors/control_api.go

```diff
@@ -189,3 +239,36 @@ func (api *ControlAPI) handleUndeploy(m *nats.Msg) {
 func (api *ControlAPI) handleWorkloadPing(m *nats.Msg) {
 	// TODO
 }
+
+type Envelope struct {
+	PayloadType string      `json:"type"`
+	Data        interface{} `json:"data,omitempty"`
+	Error       *string     `json:"error,omitempty"`
+	Code        int         `json:"code"`
+}
+
+func newEnvelope(dataType string, data interface{}, code int, err *string) Envelope {
+	return Envelope{
+		PayloadType: dataType,
+		Data:        data,
+		Error:       err,
+	}
+}
+
+func failEnvelope(dataType string, code int, err string) Envelope {
+	return Envelope{
+		PayloadType: dataType,
+		Code:        code,
+		Error:       &err,
+	}
+}
+
+func respondFail(m *nats.Msg, dataType string, code int, err string) {
```

Comment:
I dont love having these separate envelope functions.  Can we condense them to a `responseEnvelope(dataType string, code int, response any, err error)` and just have the one function?  `Data` and `Error` are already `omitempty` fields, so this just simplifies/unifies some things

Reply:
I like having them named like this because:

```
respondFail(m, MyType, 404, e)
```
is easier to quickly grasp than:
```
respondEnvelope(m, MyType, 200, nil)
```

I guess if we stick to html response code conventions it isn't that much more difficult to read.